src/vs/workbench/contrib/terminal/browser/links/terminalLinkOpeners.ts

```diff
@@ -0,0 +1,253 @@
+			if (text.substring(0, folder.name.length + 1) === folder.name + pathSeparator) {
+				text = text.substring(folder.name.length + 1);
+				return;
+			}
+		});
+		let matchLink = text;
+		if (this._capabilities.has(TerminalCapability.CommandDetection)) {
+			matchLink = this._updateLinkWithRelativeCwd(link.bufferRange.start.y, text, pathSeparator) || text;
+		}
+		const sanitizedLink = matchLink.replace(/:\d+(:\d+)?$/, '');
+		try {
+			const exactMatch = await this._getExactMatch(sanitizedLink, matchLink);
+			if (exactMatch) {
+				// If there was exactly one match, open it
+				const match = matchLink.match(/:(\d+)?(:(\d+))?$/);
+				const startLineNumber = match?.[1];
+				const startColumn = match?.[3];
+				await this._editorService.openEditor({
+					resource: exactMatch,
+					options: {
+						pinned: true,
+						revealIfOpened: true,
+						selection: startLineNumber ? {
+							startLineNumber: parseInt(startLineNumber),
+							startColumn: startColumn ? parseInt(startColumn) : 0
+						} : undefined
+					}
+				});
+				return;
+			}
+		} catch {
+			// Fallback to searching quick access
+			return this._quickInputService.quickAccess.show(text);
+		}
+		// Fallback to searching quick access
+		return this._quickInputService.quickAccess.show(text);
+	}
+
+	/*
+	* For shells with the CwdDetection capability, the cwd relative to the line
+	* of the particular link is used to narrow down the result for an exact file match, if possible.
+	*/
+	private _updateLinkWithRelativeCwd(y: number, text: string, pathSeparator: string): string | undefined {
+		const cwd = this._capabilities.get(TerminalCapability.CommandDetection)?.getCwdForLine(y);
+		if (!cwd) {
+			return undefined;
+		}
+		if (!text.includes(pathSeparator)) {
+			text = cwd + pathSeparator + text;
+		} else {
+			let commonDirs = 0;
+			let i = 0;
+			const cwdPath = cwd.split(pathSeparator).reverse();
+			const linkPath = text.split(pathSeparator);
+			while (i < cwdPath.length) {
+				if (cwdPath[i] === linkPath[i]) {
+					commonDirs++;
+				}
+				i++;
+			}
+			text = cwd + pathSeparator + linkPath.slice(commonDirs).join(pathSeparator);
+		}
+		return text;
+	}
+
+	private async _getExactMatch(sanitizedLink: string, link: string): Promise<URI | undefined> {
+		let exactResource: URI | undefined;
+		if (isAbsolute(sanitizedLink)) {
+			const scheme = this._workbenchEnvironmentService.remoteAuthority ? Schemas.vscodeRemote : Schemas.file;
+			const resource = URI.from({ scheme, path: sanitizedLink });
+			const fileStat = await this._fileService.resolve(resource);
+			if (fileStat.isFile) {
+				exactResource = resource;
+			}
+		}
+		if (!exactResource) {
+			const results = await this._searchService.fileSearch(
+				this._fileQueryBuilder.file(this._workspaceContextService.getWorkspace().folders, {
+					// Remove optional :row:col from the link as openEditor supports it
+					filePattern: sanitizedLink,
+					maxResults: 2
+				})
+			);
```

Comment:
shouldn't this be deferred to an opener?

Reply:
Not sure what you mean? It's inside `TerminalSearchLinkOpener`?